syntax = "proto3";

package c2;

service C2 {
  rpc Register(RegistrationRequest) returns (RegistrationResponse);
  rpc PullCommand(CommandRequest) returns (CommandResponse);
  rpc PushResult(CommandResult) returns (AckResponse);
}

message RegistrationRequest {
  string session_id = 1;
}

message RegistrationResponse {
  string status = 1;
}

message CommandRequest {
  string session_id = 1;
}

message CommandResponse {
  string session_id = 1;
  bytes encrypted_command = 2;
}

message CommandResult {
  string session_id = 1;
  bytes encrypted_result = 2;
}

message AckResponse {
  string status = 1;
}
